agent/consul/server_test.go

```diff
@@ -1344,11 +1347,13 @@ func TestServer_RPC_MetricsIntercept(t *testing.T) {
 			{Name: "errored", Value: "false"},
 			{Name: "request_type", Value: "read"},
 			{Name: "rpc_type", Value: "test"},
+			{Name: "server_role", Value: "unreported"},
```

Comment:
Shouldn't this get wired up with the test's `(*Server).IsLeader` func and thus not be `"unreported"`?